panacea/grpc_client.go

```diff
@@ -52,7 +55,29 @@ type GrpcClient struct {
 
 func NewGrpcClient(conf *config.Config) (GrpcClientI, error) {
 	log.Infof("dialing to Panacea gRPC endpoint: %s", conf.Panacea.GRPCAddr)
-	conn, err := grpc.Dial(conf.Panacea.GRPCAddr, grpc.WithTransportCredentials(insecure.NewCredentials()))
+
+	var conn *grpc.ClientConn
+	var err error
+
+	u, err := url.Parse(conf.Panacea.GRPCAddr)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse grpc addr: %w", err)
+	}
+
+	prefixLen := len(u.Scheme + "://")
+	addrBody := conf.Panacea.GRPCAddr[prefixLen:]
+
+	var creds credentials.TransportCredentials
+
+	if u.Scheme == "tcp" || u.Scheme == "http" {
+		creds = insecure.NewCredentials()
+	} else if u.Scheme == "https" {
+		creds = credentials.NewClientTLSFromCert(nil, "")
+	} else {
```

Comment:
If there is no shcema, an error will probably occur here. (e.g: medibloc.org)

Reply:
Thanks for comment! I'll add a error response when the `u.Scheme == nil` or `u.Scheme == ""`.

Reply:
Maybe you can only check `u.Schema == ""`.